test/functional/tool_wallet.py

```diff
@@ -36,8 +39,29 @@ def assert_tool_output(self, output, *args):
         assert_equal(stderr, '')
         assert_equal(stdout, output)
 
-    def run_test(self):
+    def wallet_path(self):
+        return os.path.join(self.nodes[0].datadir, "regtest", "wallets", "wallet.dat")
+
+    def wallet_shasum(self):
+        h = hashlib.sha1()
+        ba = bytearray(128*1024)
+        mv = memoryview(ba)
+        with open(self.wallet_path(), 'rb', buffering=0) as f:
+            for n in iter(lambda : f.readinto(mv), 0):
+                h.update(mv[:n])
+        return h.hexdigest()
+
+    def wallet_timestamp(self, op):
+        timestamp = os.path.getmtime(self.wallet_path())
+        self.log.info('Wallet file timestamp {0}: {1}'.format(op, timestamp))
+        return timestamp
 
+    def log_wallet_timestamp_comparison(self, new, old):
+        result = 'unchanged' if new == old else 'increased! **FIX ME**'
+        self.log.info('Wallet file timestamp {0}\n'.format(result))
```

Comment:
nit: instead of logging the timestamps and whether they changed, you might use `assert_equal` (or explicitly comment it out, if it currently fails)

Reply:
Thanks for reviewing! I added both logs and assertions with the logs serving as a sanity check for the assertions. Based on your feedback, I switched the assertions from passing ones that need to be inverted, to failing ones that need to be uncommented.